pkg/mimirtool/commands/rules.go

```diff
@@ -300,6 +309,22 @@ func (r *RuleCommand) setupFiles() error {
 		return errors.New("--namespaces and --ignored-namespaces cannot be set at the same time")
 	}
 
+	if r.NamespacesRegex != "" {
+		re, err := regexp.Compile(r.NamespacesRegex)
+		if err != nil {
+			return errors.Wrap(err, "invalid regex for namespaces-regex")
+		}
+		r.namespacesCompiledRegex = re
+	}
+
+	if r.IgnoredNamespacesRegex != "" {
+		re, err := regexp.Compile(r.IgnoredNamespacesRegex)
+		if err != nil {
+			return errors.Wrap(err, "invalid regex for ignored-namespaces-regex")
+		}
+		r.ignoredNamespacesCompiledRegex = re
+	}
```

Comment:
Instead of parsing these manually you can use kingpin's `RegexpVar` (now using `StringVar`).